apps/dashboard/app/controllers/batch_connect/session_contexts_controller.rb

```diff
@@ -36,26 +34,12 @@ def create
     # Read in context from form parameters
     @session_context.attributes = session_contexts_param
 
-    if params[:template].present?
-      attrib = session_contexts_param.clone
-      @session_context.attributes.each do |k, v|
-        index = @app.attributes.find_index {|a| a.id == k }
-        if index >= 0 && @app.attributes[index].opts[:disable_template]
-          attrib.delete(k)
-        end
-      end
-      @app.prefill_templates_root.tap do |p|
-        p.mkpath unless p.exist?
-      end.join(params[:template].to_s + '.json').write(JSON.pretty_generate attrib.as_json)
-    end
-
-    set_prefill_templates
-
     @session = BatchConnect::Session.new
     respond_to do |format|
       if @session.save(app: @app, context: @session_context, format: @render_format)
         cache_file.write(@session_context.to_json)  # save context to cache file
-    
+        save_template
```

Comment:
We will also have to `set_prefill_templates` here, which is why the test cases fail.

Reply:
By the way, would it be better to save the template even though the launch fails? I can see you change the behaviour into saving the template only when the launch succeeds. I actually prefer the old way since the user has already ticked the box and selected the template name, and I think there's no reason not to save it.

Reply:
> By the way, would it be better to save the template even though the launch fails? I can see you change the behaviour into saving the template only when the launch succeeds. I actually prefer the old way since the user has already ticked the box and selected the template name, and I think there's no reason not to save it.

If the launch fails - that suggests the settings were incorrect somehow. Say you've picked the wrong account and it failed to submit because the account is invalid. That saved template is now useless because won't work.

This was my thinking anyhow.

Reply:
I don't think this is the right behavior (for both issues when to save and excluding the ability to prefill things) - but I can acquiesce if you feel strongly about it.

Reply:
Which is to say - I can merge this with this behavior, subject to how many complaints we get about it lol.

Reply:
> I don't think this is the right behavior (for both issues when to save and excluding the ability to prefill things)
> Which is to say - I can merge this with this behavior, subject to how many complaints we get about it lol.

If you insist on that still, I'm fine to keep your behavior.